client/txn.go

```diff
@@ -288,15 +283,19 @@ func (txn *Txn) exec(retryable func(txn *Txn) error) error {
 			}
 		}
 		if restartErr, ok := err.(proto.TransactionRestartError); ok {
+			if log.V(2) {
+				log.Warning(err)
+			}
 			if restartErr.CanRestartTransaction() == proto.TransactionRestart_IMMEDIATE {
-				return retry.Reset, err
+				r.Reset()
+				continue
 			} else if restartErr.CanRestartTransaction() == proto.TransactionRestart_BACKOFF {
-				return retry.Continue, err
+				continue
 			}
-			// By default, fall through and return Break.
+			// By default, fall through and break.
 		}
-		return retry.Break, err
-	})
+		break
+	}
 	if err != nil && txn.haveTxnWrite {
```

Comment:
eh, that seems minor, and makes it harder to review this change. Let's leave stylistic changes like that for a future PR